### PR TITLE
Feature/feature crud pet meal

### PR DESCRIPTION
**Pull request by marclll**

### CODE REVIEW

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    public List<Meal> getAllMealData(String owner, String petName) throws ExecutionException, InterruptedException {
+        taskManager = new TaskManager();
+        taskManager.setTaskId(GET);
+        StringBuilder response = taskManager.execute(BASE_URL + owner + "/" + petName).get();
+        String jsonArray = response.substring(1, response.length()-1);
+        String[] meals = jsonArray.split(",\\{");
+        List<Meal> mealList = new ArrayList<>();
+        mealList.add(gson.fromJson(meals[0], Meal.class));
+        for (int i=0; i < meals.length; i++) {
+            meals[i] = "{" + meals[i];
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java:94]: Codacy Issue found: [The String "{" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 13
           

Complexity increasing per file
==============================
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Date.java  1
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/MealData.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Meal.java  2
         

Clones added
============
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325&bid=17392711)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    }
+
+    /**
+     * Gets a meal identified by its pet and date.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param date Date the meal was eaten
+     * @return The MealData identified by the data
+     */
+    public MealData getMealData(String owner, String petName, Date date) throws ExecutionException, InterruptedException {
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java:72]: Codacy Issue found: [Line is longer than 120 characters (found 122).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Date.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        this.month = month;
+        this.day = day;
+        this.hour = hour;
+        this.minutes = minutes;
+        this.seconds = seconds;
+    }
+
+    @NonNull
+    @Override
+    public String toString(){
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Date.java:33]: Codacy Issue found: ['{' is not preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Date.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        return year + "-" + month + "-" + day + "T" + hour + ":" + minutes + ":" + seconds;
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Date.java:34]: Codacy Issue found: [The String ":" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Meal.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+            + "name='" + date.toString() + '\''
+            + ", body=" + body
+            + '}';
+    }
+
+    @Override
+    public boolean equals(@Nullable Object obj) {
+        if (obj instanceof Meal) {
+            return ((Meal) obj).getDate().equals(this.getDate())
+                && ((Meal) obj).getBody()==(this.getBody());
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Meal.java:61]: Codacy Issue found: ['==' is not preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+                                                 String finalDate) throws ExecutionException, InterruptedException {
+        taskManager = new TaskManager();
+        taskManager.setTaskId(GET);
+        StringBuilder response =
+            taskManager.execute(BASE_URL + owner + "/" + petName + "/" + initialDate + "/" + finalDate).get();
+        String jsonArray = response.substring(1, response.length()-1);
+        String[] meals = jsonArray.split(",\\{");
+        List<Meal> mealList = new ArrayList<>();
+        mealList.add(gson.fromJson(meals[0], Meal.class));
+        for (int i=0; i < meals.length; i++) {
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java:118]: Codacy Issue found: ['=' is not preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Date.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+public class Date {
+    private int year;
+    private int month;
+    private int day;
+    private int hour;
+    private int minutes;
+    private int seconds;
+
+    /**
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Date.java:13]: Codacy Issue found: [First sentence should end with a period.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+public class MealManagerClient {
+    public static final String MEALNAME = "mealName";
+    public static final String KCAL = "kcal";
+    private static final String BASE_URL = "https://pes-my-pet-care.herokuapp.com/meal/";
+    private static final int POST = 0;
+    private static final int GET = 1;
+    private static final int DELETE = 2;
+    private static final int PUT = 3;
+    private static final Gson gson = new Gson();
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java:24]: Codacy Issue found: [Name 'gson' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Meal.java [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+    /**
+     * Returns the Meal Data
+     * @return Meal Data
+     */
+    public MealData getBody() {
+        return body;
+    }
+
+    /**
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Meal.java:39]: Codacy Issue found: [First sentence should end with a period.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/MealData.java [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+
+    /**
+     * Returns the amount of kcalories
+     * @return Amount of kcalories
+     */
+    public Double getKcal() {
+        return kcal;
+    }
+
+    /**
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/MealData.java:44]: Codacy Issue found: [First sentence should end with a period.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param meal The meal entity that contains the attributes of the meal eaten by the pet
+     */
+    public void createMeal(String owner, String petName, Meal meal) {
+        JSONObject reqJson = meal.getBody().buildMealJson();
+        taskManager = new TaskManager();
+        taskManager.setTaskId(POST);
+        taskManager.setReqBody(reqJson);
+        taskManager.execute(BASE_URL + owner + "/" + petName + "/" + meal.getDate());
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java:39]: Codacy Issue found: [The String "/" appears 14 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+     * @param finalDate Final Date
+     * @return The List containing all the meals eaten by the pet in the specified time
+     */
+    public List<Meal> getAllMealsBetween(String owner, String petName, String initialDate,
+                                                 String finalDate) throws ExecutionException, InterruptedException {
+        taskManager = new TaskManager();
+        taskManager.setTaskId(GET);
+        StringBuilder response =
+            taskManager.execute(BASE_URL + owner + "/" + petName + "/" + initialDate + "/" + finalDate).get();
+        String jsonArray = response.substring(1, response.length()-1);
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java:114]: Codacy Issue found: ['-' is not preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Date.java [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+     * Constructs a Date with the specified parameters.
```
Comment (AlbertPG) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Date.java]: Constructors doesn't have to be commented

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java [hunk 14]
```diff
@@ -0,0 +1,148 @@
...
+     * @return The List containing all the meals eaten by the pet in the specified time
+     */
+    public List<Meal> getAllMealsBetween(String owner, String petName, DateTime initialDate,
+                                                 DateTime finalDate) throws ExecutionException, InterruptedException {
+        taskManager = new TaskManager();
+        taskManager.setTaskId(GET);
+        StringBuilder response =
+            taskManager.execute(BASE_URL + owner + "/" + petName + "/between/" + initialDate + "/" + finalDate).get();
+        List<Meal> mealList = new ArrayList<>();
+        if (response.length() > 2){
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java:118]: Codacy Issue found: ['{' is not preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           

Complexity increasing per file
==============================
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java  3
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/DateTime.java  6
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Meal.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/MealData.java  2
         

Clones added
============
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325&bid=17392711)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java [hunk 15]
```diff
@@ -0,0 +1,148 @@
...
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param meal The meal entity that contains the attributes of the meal eaten by the pet
+     */
+    public void createMeal(String owner, String petName, Meal meal) {
+        JSONObject reqJson = meal.getBody().buildMealJson();
+        taskManager = new TaskManager();
+        taskManager.setTaskId(POST);
+        taskManager.setReqBody(reqJson);
+        taskManager.execute(BASE_URL + owner + "/" + petName + "/" + meal.getDate());
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java:39]: Codacy Issue found: [The String "/" appears 14 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java [hunk 16]
```diff
@@ -0,0 +1,148 @@
...
+
+public class MealManagerClient {
+    public static final String MEALNAME = "mealName";
+    public static final String KCAL = "kcal";
+    private static final String BASE_URL = "https://pes-my-pet-care.herokuapp.com/meal/";
+    private static final int POST = 0;
+    private static final int GET = 1;
+    private static final int DELETE = 2;
+    private static final int PUT = 3;
+    private static Gson GSON = new Gson();
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java:24]: Codacy Issue found: [Name 'GSON' must match pattern '^[a-z][a-zA-Z0-9]*$'.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java [hunk 17]
```diff
@@ -0,0 +1,148 @@
...
+        taskManager = new TaskManager();
+        taskManager.setTaskId(GET);
+        StringBuilder response = taskManager.execute(BASE_URL + owner + "/" + petName).get();
+        List<Meal> mealList = new ArrayList<>();
+        if (response.length() > 2) {
+            String jsonArray = response.substring(1, response.length() - 1);
+            String[] meals = jsonArray.split(",\\{");
+            mealList.add(GSON.fromJson(meals[0], Meal.class));
+            for (int i = 1; i < meals.length; i++) {
+                meals[i] = "{" + meals[i];
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java:96]: Codacy Issue found: [The String "{" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java [hunk 18]
```diff
@@ -0,0 +1,148 @@
...
+     * @return The List containing all the meals from the pet
+     */
+    public List<Meal> getAllMealData(String owner, String petName) throws ExecutionException, InterruptedException {
+        taskManager = new TaskManager();
+        taskManager.setTaskId(GET);
+        StringBuilder response = taskManager.execute(BASE_URL + owner + "/" + petName).get();
+        List<Meal> mealList = new ArrayList<>();
+        if (response.length() > 2) {
+            String jsonArray = response.substring(1, response.length() - 1);
+            String[] meals = jsonArray.split(",\\{");
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java:93]: Codacy Issue found: [The String ",\\{" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/DateTime.java [hunk 19]
```diff
@@ -0,0 +1,216 @@
...
+     * Method responsible for checking whether a year is a leap year or not.
+     * @param year The year that we want to check
+     * @return True if the given year is a leap year or false otherwise
+     */
+    private boolean isLeapYear(int year) {
+        return year % LEAP_YEAR_FREQ == 0 && (year % TWO_LAST_DIGITS != 0
+            || (year / TWO_LAST_DIGITS) % LEAP_YEAR_FREQ == 0);
+    }
+
+    @Override
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/DateTime.java:122]: Codacy Issue found: [NCSS for this method is 19 (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/DateTime.java [hunk 20]
```diff
@@ -0,0 +1,216 @@
...
+        return isYearNegative || isDateNegative;
+    }
+
+    /**
+     * Method responsible for obtaining the number of days for a given month of a given year.
+     * @param year The year value
+     * @param month The month for which we want to obtain the number of days
+     * @return The number of days of the month for that given year
+     */
+    private int numberOfDays(int year, int month) {
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/DateTime.java:93]: Codacy Issue found: [NCSS for this method is 16 (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/DateTime.java [hunk 21]
```diff
@@ -0,0 +1,216 @@
...
+    private static final int MIN_END_POS = 16;
+    private static final int SEC_START_POS = 17;
+    private int year;
+    private int month;
+    private int day;
+    private int hour;
+    private int minutes;
+    private int seconds;
+
+    public DateTime(){}
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/DateTime.java:33]: Codacy Issue found: ['}' is not preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           

Complexity increasing per file
==============================
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/DateTime.java  6
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Meal.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/MealData.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java  3
         

Clones added
============
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325&bid=17392711)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java (hunk 18 above)
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java:93]: Codacy Issue found: [The String ",\\{" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/DateTime.java [hunk 22]
```diff
@@ -0,0 +1,218 @@
...
+     * Method responsible for checking whether a year is a leap year or not.
+     * @param year The year that we want to check
+     * @return True if the given year is a leap year or false otherwise
+     */
+    private boolean isLeapYear(int year) {
+        return year % LEAP_YEAR_FREQ == 0 && (year % TWO_LAST_DIGITS != 0
+            || (year / TWO_LAST_DIGITS) % LEAP_YEAR_FREQ == 0);
+    }
+
+    @Override
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/DateTime.java:124]: Codacy Issue found: [NCSS for this method is 19 (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java (hunk 17 above)
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java:96]: Codacy Issue found: [The String "{" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/DateTime.java [hunk 23]
```diff
@@ -0,0 +1,218 @@
...
+        return isYearNegative || isDateNegative;
+    }
+
+    /**
+     * Method responsible for obtaining the number of days for a given month of a given year.
+     * @param year The year value
+     * @param month The month for which we want to obtain the number of days
+     * @return The number of days of the month for that given year
+     */
+    private int numberOfDays(int year, int month) {
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/DateTime.java:95]: Codacy Issue found: [NCSS for this method is 16 (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java (hunk 16 above)
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java:24]: Codacy Issue found: [Name 'GSON' must match pattern '^[a-z][a-zA-Z0-9]*$'.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java (hunk 15 above)
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java:39]: Codacy Issue found: [The String "/" appears 14 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Meal.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/MealData.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java  3
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/DateTime.java  6
         

Clones added
============
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325&bid=17392711)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java (hunk 17 above)
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java:96]: Codacy Issue found: [The String "{" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/DateTime.java (hunk 22 above)
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/DateTime.java:124]: Codacy Issue found: [NCSS for this method is 19 (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java (hunk 18 above)
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java:93]: Codacy Issue found: [The String ",\\{" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java (hunk 15 above)
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java:39]: Codacy Issue found: [The String "/" appears 14 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Meal.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java  3
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/MealData.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/DateTime.java  6
         

Clones added
============
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325&bid=17392711)

**Pull request merged**